From ba4c4788f268f0b31f7079332447ebf6ce4fa568 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Mon, 16 Jul 2018 07:36:14 +0200
Subject: [PATCH 1/1] test: tests for utf_to_lower() utf_to_upper().

Provide unit tests for utf_to_lower() utf_to_upper().

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 test/unicode_ut.c | 38 ++++++++++++++++++++++++++++++++++++++
 1 file changed, 38 insertions(+)

diff --git a/test/unicode_ut.c b/test/unicode_ut.c
index 96b4852c0cd..488e1df6daf 100644
--- a/test/unicode_ut.c
+++ b/test/unicode_ut.c
@@ -248,6 +248,42 @@ int ut_utf16_utf8_strnlen(void)
 	return 0;
 }
 
+int ut_utf_to_lower(void)
+{
+	if (utf_to_lower('@') != '@')
+		return 1;
+	if (utf_to_lower('A') != 'a')
+		return 1;
+	if (utf_to_lower('Z') != 'z')
+		return 1;
+	if (utf_to_lower('[') != '[')
+		return 1;
+	if (utf_to_lower('m') != 'm')
+		return 1;
+	/* Cyrillic letter I*/
+	if (utf_to_lower(0x0418) != 0x0438)
+		return 0;
+	return 0;
+}
+
+int ut_utf_to_upper(void)
+{
+	if (utf_to_upper('M') != 'M')
+		return 1;
+	if (utf_to_lower('`') != '`')
+		return 1;
+	if (utf_to_upper('a') != 'A')
+		return 1;
+	if (utf_to_upper('z') != 'Z')
+		return 1;
+	if (utf_to_upper('{') != '{')
+		return 1;
+	/* Cyrillic letter I */
+	if (utf_to_upper(0x0438) != 0x0418)
+		return 1;
+	return 0;
+}
+
 int do_ut_unicode(cmd_tbl_t *cmdtp, int flag, int argc, char * const argv[])
 {
 	int ret = 0;
@@ -260,6 +296,8 @@ int do_ut_unicode(cmd_tbl_t *cmdtp, int flag, int argc, char * const argv[])
 	ret |= ut_utf16_put();
 	ret |= ut_utf16_utf8_strlen();
 	ret |= ut_utf16_utf8_strnlen();
+	ret |= ut_utf_to_lower();
+	ret |= ut_utf_to_upper();
 
 	printf("Test %s\n", ret ? "failed" : "passed");
 
-- 
2.18.0

